buildbot/master/files/config/factories.py

```diff
@@ -103,6 +103,86 @@ def make_step(self, command):
         return step_class(**step_kwargs)
 
 
+class StepsYAMLParsingStep(buildstep.BuildStep):
+    """
+    Step which resolves the in-tree YAML and dynamically add test steps
+    """
+    yaml_path = ""
+
+    def __init__(self, builder_name, environment, yaml_path):
+        print(yaml_path)
+        self.builder_name = builder_name
+        self.environment = environment
+        self.yaml_path = yaml_path
+
+    def run(self):
+        print(self.yaml_path)
+        try:
+            full_yaml_path = os.path.join(self.build.workdir, self.yaml_path)
+            with open(full_yaml_path) as steps_file:
+                builder_steps = yaml.safe_load(steps_file)
+            commands = builder_steps[self.builder_name]
+            dynamic_steps = [self.make_step(command) for command in commands]
+        except Exception as e:  # Bad step configuration, fail build
+            print(str(e))
+            dynamic_steps = [BadConfigurationStep(e)]
+
+        # TODO: windows compatibility (use a custom script for this?)
+        pkill_step = [steps.ShellCommand(command=["pkill", "-x", "servo"],
```

Comment:
Let's not put this step in a list here, since `pkill_step` is singular.

Instead, we can wrap it in a list by adding a line here:
```
static_steps = [pkill_step]
```
and later on using `static_steps + dynamic_steps`.

(We could also directly use `[pkill_step] + dynamic_steps`, but I think that's a little less clear.)